### PR TITLE
fix: try catch out of gas edge case

### PR DESCRIPTION
**Pull request by 8sunyuan**

### CODE REVIEW

File: script/releases/v1.0.2-slashing/1-eoa.s.sol [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    /// Note: The upgrade script can call this with `areMatching == true` to check that these impl
+    /// addresses _are_ matches.
+    function _validateNewImplAddresses(bool areMatching) internal view {
+        function (address, address, string memory) internal pure assertion =
+            areMatching ? _assertMatch : _assertNotMatch;
+
+
+        assertion(
+            _getProxyImpl(address(Env.proxy.strategyManager())),
+            address(Env.impl.delegationManager()),
```
Comment (gpsanant) [inline script/releases/v1.0.2-slashing/1-eoa.s.sol:49]:
```suggestion
            address(Env.impl.strategyManager()),
```

Comment (8sunyuan) [reply script/releases/v1.0.2-slashing/1-eoa.s.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/971/commits/4433cb11be21a2ed0698eeeb418cb4ef76c6e0e3

File: src/test/unit/StrategyManagerUnit.t.sol [hunk 2]
```diff
@@ -1517,32 +1518,32 @@ contract StrategyManagerUnitTests_burnShares is StrategyManagerUnitTests {
...
+
+    //     uint256 strategyBalanceBefore = token.balanceOf(address(strategy));
+    //     uint256 burnAddressBalanceBefore = token.balanceOf(strategyManager.DEFAULT_BURN_ADDRESS());
+    //     cheats.prank(address(delegationManagerMock));
+    //     strategyManager.burnShares(strategy, sharesToBurn);
+    //     uint256 strategyBalanceAfter = token.balanceOf(address(strategy));
+    //     uint256 burnAddressBalanceAfter = token.balanceOf(strategyManager.DEFAULT_BURN_ADDRESS());
+
+    //     assertEq(burnAddressBalanceBefore, burnAddressBalanceAfter, "burnAddressBalanceBefore != burnAddressBalanceAfter");
+    //     assertEq(strategyBalanceBefore, strategyBalanceAfter, "strategyBalanceBefore != strategyBalanceAfter");
```
Comment (gpsanant) [inline src/test/unit/StrategyManagerUnit.t.sol:1545]: can we make sure this test fails?

Comment (ypatil12) [reply src/test/unit/StrategyManagerUnit.t.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/971/commits/6a880aa03eb8b0b065dd6226c22dabfe4860e0b0

**Review by gpsanant — APPROVED** [review]

**Pull request merged**